docs/authoring_test_cases.md

```diff
@@ -1,11 +1,83 @@
+  tests in the suite will be run only when the `mode` specified to the runner on the command line is `client`. If not 
+  specified in a suite, tests are run regardless of the command line `mode`.
+
+* `relevantProtocols` is used to limit tests only to a specific protocol, such as **Connect**, **gRPC**, or **gRPC-web**. If not
+  specified, tests are run for all protocols.
+
+* `relevantHttpVersions` is used to limit tests to certain HTTP versions, such as **HTTP 1.1**, **HTTP/2**, or **HTTP/3**. If not
+  specified, tests are run for all HTTP versions.
+
+* `relevantCodecs` is used to limit tests to certain codec formats, such as **JSON** or **binary**. If not specified, tests are
+   run for all codecs.
+
+* `relevantCompressions` is used to limit tests to certain compression algorithms, such as **gzip**, **brotli**, or **snappy**. If not
+  specified, tests are run for all compressions.
+
+* `connectVersionMode` allows you to either require or ignore validation of the Connect version header or query param. This
+  should be left unspecified if the suite is agnostic to this validation behavior.
+
+The below `reliesOn` directives are used to signal to the test runner how the reference client or server should be
+configured when running tests:
+
+* `reliesOnTls` specifies that a suite relies on TLS. Defaults to `false`.
+
+* `reliesOnTlsClientCerts` specifies that the suite relies on the _client_ using TLS certificates to authenticate with 
+  the server. Note that if this is set to `true`, `reliesOnTls` must also be `true`. Defaults to `false`.
+
+* `reliesOnConnectGet` specifies that the suite relies on the Connect GET protocol. Defaults to `false`.
+
+* `reliesOnMessageReceiveLimit` specifies that the suite relies on support for limiting the size of received messages.
+  When `true`, the `mode` property must be set to indicate whether the client or server should support the limit. Defaults
+  to `false`.
+
+## Test cases
+
+Test cases are specified in the `testCases` property of the suite. Each test case starts with the `request` property 
+which defines the request which will be sent to a client during the test run. Each `request` must specify the following
+fields:
+
+* `testName` - For naming conventions, see [below](#naming-conventions).
+* `service` - The fully-qualified path of the service this test will interact with, typically the `ConformanceService`.
+* `method` - This is a string specifying the method on `service` that will be called.
+* `streamType` - One of `STREAM_TYPE_UNARY`, `STREAM_TYPE_CLIENT_STREAM`, `STREAM_TYPE_SERVER_STREAM`, 
+  `STREAM_TYPE_HALF_DUPLEX_BIDI_STREAM`, or `STREAM_TYPE_FULL_DUPLEX_BIDI_STREAM`.
+
+Once the above are specified, you can then define your request. For a full list of fields to specify in the request,
+see the [`ClientCompatRequest`][client-compat-proto] message in the Conformance Protobuf definitions.
```

Comment:
I think it will be more helpful to link to the actual message description in the generated docs instead of linking to the source of the containing file
https://buf.build/connectrpc/conformance/docs/main:connectrpc.conformance.v1#connectrpc.conformance.v1.ClientCompatRequest